sql/parser/eval.go

```diff
@@ -401,14 +401,33 @@ var cmpOps = map[cmpArgs]func(Datum, Datum) (DBool, error){
 
 	cmpArgs{Like, stringType, stringType}: func(left Datum, right Datum) (DBool, error) {
 		pattern := regexp.QuoteMeta(string(right.(DString)))
+		pattern = fmt.Sprintf("^%s$", pattern)
+		// Replace LIKE specific wildcards with standard wildcards
 		pattern = strings.Replace(pattern, "%", ".*", -1)
 		pattern = strings.Replace(pattern, "_", ".", -1)
-		pattern = fmt.Sprintf("^%s$", pattern)
 
 		// TODO(pmattis): Can we cache this regexp compilation?
 		re := regexp.MustCompile(pattern)
 		return DBool(re.MatchString(string(left.(DString)))), nil
 	},
+
+	cmpArgs{SimilarTo, stringType, stringType}: func(left Datum, right Datum) (DBool, error) {
+		pattern := string(right.(DString))
+		pattern = fmt.Sprintf("^%s$", pattern)
+		// Escape all periods, because they are NOT metacharacters for SQL-style regexp
+		pattern = strings.Replace(pattern, ".", "\\.", -1)
+		// Replace SQL-style regexp specific wildcards with standard regexp wildcards
+		pattern = strings.Replace(pattern, "%", ".*", -1)
+		pattern = strings.Replace(pattern, "_", ".", -1)
```

Comment:
You may want to derive inspiration from [similar_escape](https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/regexp.c#L658).